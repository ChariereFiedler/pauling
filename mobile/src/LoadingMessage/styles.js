/* @flow */
import { StyleSheet } from 'react-native';

import { colors } from 'app/settings';

export default StyleSheet.create({
  LoadingMessage: {
    marginTop: 50,
    padding: 20,
  },
  Text: {
    color: colors.textSecondaryColor,
    textAlign: 'center',
  },
});
