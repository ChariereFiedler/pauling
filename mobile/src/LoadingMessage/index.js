/* @flow */
import React from 'react';
import { View } from 'react-native';
import { Spinner, Text } from 'native-base';

import { colors } from 'app/settings';
import styles from './styles';

type Props = {|
  children?: string,
|};

const LoadingMessage = ({ children }: Props) =>
  <View style={styles.LoadingMessage}>
    <Text style={styles.Text}>
      {children}
    </Text>
    <Spinner color={colors.textSecondaryColor} />
  </View>;

export default LoadingMessage;
