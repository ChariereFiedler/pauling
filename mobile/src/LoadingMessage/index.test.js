import React from 'react';
import { shallow } from 'enzyme';

import LoadingMessage from 'app/LoadingMessage';

describe('LoadingMessage', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <LoadingMessage>
        Some content to explain why it is loading.
      </LoadingMessage>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
