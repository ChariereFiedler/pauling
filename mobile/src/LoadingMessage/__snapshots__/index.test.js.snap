// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingMessage renders correctly 1`] = `
<Component
  style={
    Object {
      "marginTop": 50,
      "padding": 20,
    }
  }
>
  <Styled(Text)
    style={
      Object {
        "color": "#777",
        "textAlign": "center",
      }
    }
  >
    Some content to explain why it is loading.
  </Styled(Text)>
  <Styled(Spinner)
    color="#777"
  />
</Component>
`;
